test/test_e2e.rb

```diff
@@ -104,4 +109,27 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert_equal 1, queue.fail_fast
+    assert_operator queue.unprocessed_jobs.length, :<=, 3
+    assert_operator queue.processed_jobs.length, :<=, 2
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert_equal queue.fail_fast, queue.example_failures.length +
+                                  queue.non_example_errors.length +
+                                  queue.total_requeues
+  end
+
+  def test_suite_with_failures_and_no_fail_fast
+    queue = exec_build("failing_suite")
+
+    assert queue.fail_fast.zero?
+    assert queue.unprocessed_jobs.empty?
+    refute queue.build_failed_fast?
+    refute queue.build_successful?
+    refute queue.processed_jobs.empty?
```

Comment:
These assertions should be incorporated in the existing `test_suite_with_legit_failures` test. We don't have to re-run the same  build.